x-pack/plugins/ml/public/application/explorer/anomalies_map.tsx

```diff
@@ -207,24 +204,26 @@ export const AnomaliesMap: FC<Props> = ({ anomalies, jobIds }) => {
     [...jobIds]
   );
 
-  const layerList = useMemo(() => {
-    let layers: VectorLayerDescriptor[] = [];
-    // Loop through suggestions list and make a layer for each
-    if (EMSSuggestions?.length) {
-      let count = 0;
-      layers = EMSSuggestions.reduce(function (result, suggestion) {
-        if (suggestion) {
-          const visible = count === 0;
-          result.push(getChoroplethAnomaliesLayer(anomalies, suggestion, visible));
-          count++;
-        }
-        return result;
-      }, [] as VectorLayerDescriptor[]);
-    }
-    return layers;
+  const layerList: VectorLayerDescriptor[] = useMemo(() => {
+    if (!EMSSuggestions?.length) return [];
+
+    return EMSSuggestions.map((suggestion) => {
+      return getChoroplethAnomaliesLayer(anomalies, suggestion);
+    }, [] as VectorLayerDescriptor[]);
   }, [EMSSuggestions, anomalies]);
 
-  if (EMSSuggestions?.length === 0) {
+  const layersWithAnomalies = layerList.filter((layer) => {
+    // @ts-ignore _rows does not exist - can remove when VectorLayerDescriptor is updated
+    const rows = layer.joins ? layer.joins[0].right.__rows : [];
+    return rows.length;
+  });
+
+  // set the layer with anomalies to visible
+  if (layersWithAnomalies.length > 0) {
+    layersWithAnomalies[0].visible = true;
+  }
+
+  if (EMSSuggestions?.length === 0 || !(layersWithAnomalies.length > 0)) {
```

Comment:
I think it's easier to follow the condition this way 
```suggestion
  if (EMSSuggestions?.length === 0 || layersWithAnomalies.length === 0) {
```